### PR TITLE
ICMSLST-675 Move and add User Certificate to SIL

### PR DESCRIPTION
**Pull request by kevinetienne**

### CODE REVIEW

File: web/domains/case/_import/fa/urls.py [hunk 1]
```diff
@@ -78,6 +78,35 @@
                         ]
                     ),
                 ),
+                path(
+                    "certificates/",
+                    include(
+                        [
+                            path("manage/", views.list_certificates, name="list-certificates"),
```
Comment (oskusalerma) [inline web/domains/case/_import/fa/urls.py:85]: can we choose a consistent name for this. either list or manage, not both.

File: web/domains/case/_import/fa/urls.py [hunk 2]
```diff
@@ -78,6 +78,35 @@
...
+                            path("create/", views.create_certificate, name="create-certificate"),
+                            path(
+                                "<int:document_pk>/view/",
```
Comment (oskusalerma) [inline web/domains/case/_import/fa/urls.py:88]: this shuold be under the grouping below

File: web/domains/case/_import/fa/urls.py [hunk 3]
```diff
@@ -78,6 +78,35 @@
...
+                                views.view_certificate_document,
+                                name="view-certificate-document",
+                            ),
+                            path(
+                                "<int:certificate_pk>/",
+                                include(
+                                    [
+                                        path(
+                                            "view/", views.edit_certificate, name="edit-certificate"
```
Comment (oskusalerma) [inline web/domains/case/_import/fa/urls.py:97]: this is edit, not view

File: web/domains/case/_import/fa/views.py [hunk 4]
```diff
@@ -390,6 +393,154 @@ def edit_import_contact(
...
+@login_required
+def edit_certificate(
+    request: HttpRequest, *, application_pk: int, certificate_pk: int
+) -> HttpResponse:
+    with transaction.atomic():
+        application = get_object_or_404(
+            ImportApplication.objects.select_for_update(), pk=application_pk
+        )
+        check_application_permission(application, request.user, "import")
+        certificate = get_object_or_404(UserImportCertificate, pk=certificate_pk)
```
Comment (oskusalerma) [inline web/domains/case/_import/fa/views.py:476]: can we get the certificate from application.XXX so we're sure it belongs to the application

File: web/domains/case/_import/fa_oil/models.py [hunk 5]
```diff
@@ -50,7 +25,7 @@ class VerifiedCertificate(models.Model):
         related_name="verified_certificates",
     )
     firearms_authority = models.ForeignKey(
-        FirearmsAuthority, on_delete=models.PROTECT, related_name="verified_certificates"
+        "web.FirearmsAuthority", on_delete=models.PROTECT, related_name="verified_certificates"
```
Comment (oskusalerma) [inline web/domains/case/_import/fa_oil/models.py:28]: see if it works without web.

File: web/domains/case/_import/fa_sil/models.py [hunk 6]
```diff
@@ -48,6 +48,11 @@ class SILApplication(ImportApplication):
     user_section5 = models.ManyToManyField(SILUserSection5, related_name="+")
     verified_section5 = models.ManyToManyField(Section5Authority, related_name="+")
 
+    # certificates
+    user_imported_certificates = models.ManyToManyField(
+        "web.UserImportCertificate", related_name="+"
```
Comment (oskusalerma) [inline web/domains/case/_import/fa_sil/models.py:53]: try without web.

File: web/migrations/0016_userimportcertificate.py [hunk 7]
```diff
@@ -56,8 +56,6 @@ class Migration(migrations.Migration):
         migrations.AddField(
             model_name="openindividuallicenceapplication",
             name="user_imported_certificates",
-            field=models.ManyToManyField(
-                related_name="import_application", to="web.UserImportCertificate"
-            ),
+            field=models.ManyToManyField(related_name="+", to="web.UserImportCertificate"),
```
Comment (oskusalerma) [inline web/migrations/0016_userimportcertificate.py:59]: try without web.

File: web/migrations/0062_add_silapplication.py [hunk 8]
```diff
@@ -79,6 +79,10 @@ class Migration(migrations.Migration):
                     "verified_section5",
                     models.ManyToManyField(related_name="+", to="web.Section5Authority"),
                 ),
+                (
+                    "user_imported_certificates",
+                    models.ManyToManyField(related_name="+", to="web.UserImportCertificate"),
```
Comment (oskusalerma) [inline web/migrations/0062_add_silapplication.py:84]: try without web.

**Review by oskusalerma — CHANGES_REQUESTED** [review]

File: web/domains/case/_import/fa_oil/models.py (hunk 5 above)
Comment (kevinetienne) [reply web/domains/case/_import/fa_oil/models.py]: It worked, updated in https://github.com/uktrade/icms/pull/417/commits/e4cd0694e57d6fb1241adf0a3beaf9986fc340b3.

File: web/domains/case/_import/fa/views.py (hunk 4 above)
Comment (kevinetienne) [reply web/domains/case/_import/fa/views.py]: Used the related relationship `user_imported_certificates` in  https://github.com/uktrade/icms/pull/417/commits/e4cd0694e57d6fb1241adf0a3beaf9986fc340b3.

File: web/domains/case/_import/fa/urls.py (hunk 3 above)
Comment (kevinetienne) [reply web/domains/case/_import/fa/urls.py]: Fixed in https://github.com/uktrade/icms/pull/417/commits/e4cd0694e57d6fb1241adf0a3beaf9986fc340b3.

File: web/domains/case/_import/fa/urls.py (hunk 2 above)
Comment (kevinetienne) [reply web/domains/case/_import/fa/urls.py]: Fixed in https://github.com/uktrade/icms/pull/417/commits/e4cd0694e57d6fb1241adf0a3beaf9986fc340b3.

File: web/domains/case/_import/fa/urls.py (hunk 1 above)
Comment (kevinetienne) [reply web/domains/case/_import/fa/urls.py]: I went with `manage` as it seems we are using manage in the urls in a few different places. See https://github.com/uktrade/icms/pull/417/commits/e4cd0694e57d6fb1241adf0a3beaf9986fc340b3.

**Review by oskusalerma — APPROVED** [review]

**Pull request merged**